packages/fuels-core/src/abi_encoder.rs

```diff
@@ -548,8 +581,16 @@ mod tests {
         assert_eq!(abi_encoder.function_selector, expected_function_selector);
     }
 
+    // The encoding follows the ABI specs defined  [here](https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/abi.md)
+    // An enum doesn't change its encoding width according to which variant it
+    // is currently holding. It is always as wide as its biggest variant. If we
+    // store a smaller variant in it, it will be left padded until it reaches
+    // the allotted enum size. Besides the value itself, an enum also stores a
+    // discriminant to indicate which variant it is currently holding.
```

Comment:
I think this is a repetition of what's being said in some other place (or should be said some other place haah). So you can remove the above part, but enrich the comment below a little

Reply:
`An enum is always as wide as its biggest variant` is the key sentence here, very well put -> should be in the below comment